queries.sql

```diff
@@ -8,4 +8,68 @@ select * from animals where name IN ('Agumon', 'Pikachu');
 select name, escape_attempts from animals where weight_kg > 10.5;
 select * from animals where neutered = true; 
 select * from animals where name !=  'Gabumon'; 
-select * from animals where weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+select * from animals where weight_kg BETWEEN 10.4 and 17.3;
+
+-- Transaction and Roll back
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'unspecified';
+
+COMMIT;
+
+-- Requirements_1: 
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' 
+WHERE name LIKE '%mon';
+
+SAVEPOINT a;
+
+UPDATE animals SET species = 'pokemon' 
+WHERE name NOT LIKE '%mon';
+
+SAVEPOINT b;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+
+
+-- Requirement-2
+
+BEGIN TRANSACTION;
+
+TRUNCATE TABLE animals;
+
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+
+-- Requirement-3 
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT delete;
+
+UPDATE animals
+SET weight_kg = -1 * weight_kg;
+
+SELECT * FROM animals;
+ROLLBACK;
+COMMIT;
+
+-- Requirement-4 
+SELECT count(name) as Number_of_Animals from animals;
+SELECT name, escape_attempts FROM animals WHERE escape_attempts = 0; 
+SELECT AVG(weight_kg) AS Weight_Average  FROM animals;
+SELECT name, neutered, escape_attempts 
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals); 
+
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+
```

Comment:
- Please re-run this query and attach the right results in your PR after correcting the data in your data.sql file. Please fix this 👍 👍
- Also, I cannot see the query for this query, `What is the average number of escape attempts per animal type of those born between 1990 and 2000?` in your query file, though I see you have included the results in your PR. Please fix this 🙏 
- Also, the attached result for the above-mentioned query is not correct, to get the `average number of escape attempts per animal type of those born between 1990 and 2000?`. Please use this query:
```sql
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
```